File: partner-service/src/main/java/com/example/partner_service/entity/Partner.java
package com.example.partner_service.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;

@Entity
@Table(name = "tblpartner")
@Getter
@Setter
@NoArgsConstructor
public class Partner {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private String company;

    @Column(nullable = false)
    private String phone;

    @Column(nullable = false, unique = true)
    private String email;

    @Column(name = "signDate", nullable = false)
    private LocalDate signDate;

    @Column(name = "expirationDate", nullable = false)
    private LocalDate expirationDate;

}

File: client/src/model/Partner.js
export class Partner {
    constructor(id = null, name = '', company = '', phone = '', email = '', signDate = null, expirationDate = null) {
        this.id = id;
        this.name = name;
        this.company = company;
        this.phone = phone;
        this.email = email;
        this.signDate = signDate;
        this.expirationDate = expirationDate;
    }
}
